connmgr.go

```diff
@@ -32,9 +32,13 @@ type BasicConnMgr struct {
 	gracePeriod time.Duration
 	peers       map[peer.ID]*peerInfo
 
+	plk       sync.RWMutex
+	protected map[peer.ID]struct{}
+
 	// channel-based semaphore that enforces only a single trim is in progress
 	trimRunningCh chan struct{}
 	lastTrim      time.Time
+	silencePeriod time.Duration
```

Comment:
To minimise test races if we ever run these tests in parallel.

Reply:
Ideally we'd modify the constructor to take a config struct, but that would be a breaking change.